solutions/hw2/task3/smart_args.py

```diff
@@ -0,0 +1,66 @@
+from typing import Callable, List, Dict, Any
+import copy
+import inspect
+
+
+class Evaluated:
+    def __init__(self, eval_func: Callable[[], Any]):
+        self.eval_func = eval_func
+        self.check_for_mixed_usage()
+
+    def check_for_mixed_usage(self):
+        if self.eval_func is Isolated:
+            raise TypeError("Can't use evaluated and isolated together")
+
+
+class Isolated:
+    pass
+
+
+def get_keyword_defaults(function) -> Dict[str, Any]:
+    signature = inspect.signature(function)
+    return {
+        k: v.default
+        for k, v in signature.parameters.items()
+        if v.default is not inspect.Parameter.empty and v.kind == inspect.Parameter.KEYWORD_ONLY
+    }
+
+
+def get_isolated_kwargs_names(keyword_defaults) -> List[str]:
+    return [key for key, value in keyword_defaults if isinstance(value, Isolated)]
```

Comment:
I think these methods should be part of the `SmartArgs` class. Since they don't use self they can be static.